src/main/kotlin/app/revanced/patches/tumblr/timelinefilter/TimelineFilterPatch.kt

```diff
@@ -28,30 +28,25 @@ object TimelineFilterPatch : BytecodePatch(
     internal lateinit var addObjectTypeFilter: (typeName: String) -> Unit private set
 
     override fun execute(context: BytecodeContext) {
-
         TimelineFilterIntegrationFingerprint.result?.let { integration ->
             val filterInsertIndex = integration.scanResult.patternScanResult!!.startIndex
 
             integration.mutableMethod.apply {
-                // This is the List.add call from the dummy object type filter
-                val instr = getInstruction<BuilderInstruction35c>(filterInsertIndex + 1)
-
-                if (instr.registerCount != 2) throw TimelineFilterIntegrationFingerprint.exception
+                val addInstruction = getInstruction<BuilderInstruction35c>(filterInsertIndex + 1)
+                if (addInstruction.registerCount != 2) throw TimelineFilterIntegrationFingerprint.exception
 
-                // From the dummy filter call, we can get the 2 registers we need to add more filters
-                val listRegister = instr.registerC
-                val stringRegister = instr.registerD
+                val filterListRegister = addInstruction.registerC
+                val stringRegister = addInstruction.registerD
 
-                // Remove "BLOCKED_OBJECT_DUMMY" object type filter
+                // Remove "BLOCKED_OBJECT_DUMMY"
                 removeInstructions(filterInsertIndex, 2)
 
                 addObjectTypeFilter = { typeName ->
-                    // The java equivalent of this is
-                    //   blockedObjectTypes.add({typeName})
+                    // blockedObjectTypes.add({typeName})
                     addInstructionsWithLabels(
                         filterInsertIndex, """
                             const-string v$stringRegister, "$typeName"
-                            invoke-interface { v$listRegister, v$stringRegister }, Ljava/util/List;->add(Ljava/lang/Object;)Z
```

Comment:
This does not work. HashSet is a class and not an interface, invoke-virtual should be used. Also because of your refactor, the dummy filter fingerprint is broken since it matches `const-string, invoke-interface` but HashSet.add gets compiled to `const-string, invoke-virtual`.

Either update the fingerprint and the injected smali to invoke-virtual, or update the integration to have variable type Set instead of HashSet (the interface that HashSet implements) and update the injected smali to `invoke-interface` Set.add